### ISSUE TITLE
Create a Qubes Builder V2 configuration linter

### ISSUE THREAD

**Issue by ben-grande** [QubesOS/qubes-issues#9046]
### The problem you're addressing (if any)

Mistakes happen, in some cases, `yamllint` can detect it: https://github.com/QubesOS/qubes-issues/issues/9043: https://github.com/QubesOS/qubes-release-configs/commit/b0a7e2fd0c44075a5b1e133f7ec7da43de92c863: *error: duplication of key "flavor" in mapping*

Sometimes it may not be able to detect it such as when using the wrong dist for a Debian version:
```yaml
- debian-12-xfce:
    dist: bullseye
```

Or when the flavor is wrong:
```yaml
- debian-12-xfce:
    flavor: minimal
```

And many other types of mistakes that can be detected. Yes, I understand that may be some amount of cases it won't detect, but a more specialized tool than `yamllint` would help to prevent some errors. Quantifying this in this stage is not possible as I didn't try to create the linter to have sample data.

### The solution you'd like

Create a linter specialized in the Qubes Builder V2 key options.
Use CI to lint these files.

The linter can be part of the module `qubesbuilder.cli`, so a `lint` option or subcommand can be passed for it to only evaluate the configuration options but do not fetch or build anything.

### The value to a user, and who that user might be

Developers who modify the Builder V2 configuration files will be able to detect problems before they are merged and released to testing as CI will notify them.
Users in `stable` will have less chance of getting a wrong build in case the users in `testing` don't detect the problem.

**Comment by ben-grande** [comment]
@fepitre Since v2 has matured, what do you think of this? Is it possible with the current structure or would a lot of changes be necessary?

**Comment by fepitre** [comment]
Yes it is enough mature now and a linter could be easy to implement.
